app.rb

```diff
@@ -31,10 +42,12 @@ def create_student
     print 'Name: '
     name = gets.chomp
     print 'Has parent permission? [Y/N]: '
-    parent_permission = gets.chomp
-    person = Student.new(name, age, parent_permission)
+    parent_permission = gets.chomp.downcase == 'y'
+    id = Random.rand(1...1000)
+    person = Student.new(name, age, parent_permission: parent_permission, id: id)
     @people.push(person)
     puts "Student '#{name}' created successfully"
+    save_people_to_json

```

Comment:
- [ ]  [OPTIONAL]
Good job, the app can now runs correctly! However, it appears to save data immediately after entering details for each section. In this activity, we actually want the app to only persist data into _json_ files on the exit of the app, thus, after _option 7_ has been selected. Let's improve upon this aspect.